cmd/sonobuoy/app/gen_test.go

```diff
@@ -244,44 +257,105 @@ func TestResolveConfig(t *testing.T) {
 				PluginSelections: []plugin.Selection{},
 				Resources:        []string{},
 			},
+			expectedGenConfigPlugins: &client.GenConfig{},
+		}, {
+			name:  "Plugins can be loaded by directory",
+			input: "--plugin testdata/testPluginDir",
+			expected: &config.Config{
+				Namespace:        "sonobuoy",
+				WorkerImage:      "gcr.io/heptio-images/sonobuoy:" + buildinfo.Version,
+				ImagePullPolicy:  "IfNotPresent",
+				PluginSelections: nil,
+				PluginSearchPath: defaultPluginSearchPath,
+				Aggregation:      defaultAggr,
+				Resources:        config.DefaultResources,
+			},
+			expectedGenConfigPlugins: &client.GenConfig{
+				StaticPlugins: []*manifest.Manifest{
+					{SonobuoyConfig: manifest.SonobuoyConfig{Driver: "Job", PluginName: "plugin1", SkipCleanup: false, ResultFormat: "raw"}, Spec: manifest.Container{Container: v1.Container{Name: "plugin", Image: "foo/bar:v1", Command: []string{"./run.sh"}, VolumeMounts: []v1.VolumeMount{{ReadOnly: false, Name: "results", MountPath: plugin.ResultsDir}}}}},
```

Comment:
These static plugin lines are long but shouldn't really change. I just had to grab one and then a bit of copy/paste. Details arent too important here, the point is that the right set of plugins got loaded (really just need to look at name/image)